addons/mail/static/src/core/common/store_service.js

```diff
@@ -90,10 +90,13 @@ export class Store extends BaseStore {
         inverse: "storeAsTrackedImStatus",
         /** @this {import("models").Store} */
         onUpdate() {
-            this.env.services["im_status"].registerToImStatus(
-                "res.partner",
-                this.imStatusTrackedPersonas.map((p) => p.id)
-            );
+            const channels = this.imStatusTrackedPersonas.map(({ type, id }) => {
+                const model = type === "partner" ? "res.partner" : "mail.guest";
+                return `odoo-presence-${model}_${id}`;
+            });
+            for (const c of channels) {
+                this.env.services.bus_service.addChannel(c);
+            }
```

Comment:
We no longer need to make this on the `many` side as we're adding them one by one, we can avoid many loops if just doing it on the `one` side?

Could also take the opportunity to `deleteChannel` when deleting the persona.